release/pkg/bundles/package-controller.go

```diff
@@ -191,3 +204,60 @@ func replaceTag(uri, tag string) string {
 	uri = strings.Join(NewURIList[:], ":")
 	return uri
 }
+
+// getTag is used to replace get the tag of an ImageURI
+func getTag(uri string) string {
+	newURIList := strings.Split(uri, ":")
+	if len(newURIList) < 2 {
+		return uri
+	}
+	return newURIList[len(newURIList)-1]
+}
+
+// recoverBundle is a function used if we have a helm error to revert the bundle back to it's original state using Git Tag version.
+func recoverBundle(r *releasetypes.ReleaseConfig, imageDigests map[string]string, helmtag, imagetag, tokentag string) (map[string]anywherev1alpha1.Image, []string) {
+	artifacts := map[string][]releasetypes.Artifact{
+		"eks-anywhere-packages": r.BundleArtifactsTable["eks-anywhere-packages"],
+		"ecr-token-refresher":   r.BundleArtifactsTable["ecr-token-refresher"],
+	}
+	sortedComponentNames := bundleutils.SortArtifactsMap(artifacts)
+	bundleImageArtifacts := map[string]anywherev1alpha1.Image{}
+	artifactHashes := []string{}
+	for _, componentName := range sortedComponentNames {
+		for _, artifact := range artifacts[componentName] {
+			if artifact.Image != nil {
+				imageArtifact := artifact.Image
+				bundleImageArtifact := anywherev1alpha1.Image{}
+				if strings.HasSuffix(imageArtifact.AssetName, "helm") {
+					imageArtifact.ReleaseImageURI = replaceTag(imageArtifact.ReleaseImageURI, helmtag)
+					Digest := imageDigests[imageArtifact.ReleaseImageURI]
+					assetName := strings.TrimSuffix(imageArtifact.AssetName, "-helm")
+					bundleImageArtifact = anywherev1alpha1.Image{
+						Name:        assetName,
+						Description: fmt.Sprintf("Helm chart for %s", assetName),
+						URI:         imageArtifact.ReleaseImageURI,
+						ImageDigest: Digest,
+					}
+				} else {
+					if strings.HasSuffix(imageArtifact.AssetName, "eks-anywhere-packages") {
```

Comment:
Do we expect it to be at the end of the asset name? Should we do `Contains` or exact match instead?

Reply:
This was so it wouldn't get matched with `eks-anywhere-packages-helm`, but I guess there is a check up above the the string `helm` so this wouldn't catch anyway?